src/rate-limiter.ts

```diff
@@ -0,0 +1,21 @@
+import { PostHogPersistence } from './posthog-persistence'
+
+export class RateLimiter {
```

Comment:
See the other comments on the other PR but I think this also needs to be per-resource (events, recordings etc.)

Reply:
To level up the fun we don't really know what we're sending

The clearest non-configurable thing I could see was that session recordings always add a `_batchKey` so I've used that.

no batch key -> events
`events` as the batch key -> events
`sessionRecordings` as the batch key -> session recordings